prime-router/src/main/kotlin/fhirengine/translation/TranslationSchemaManager.kt

```diff
@@ -0,0 +1,91 @@
+package gov.cdc.prime.router.fhirengine.translation
+
+import gov.cdc.prime.router.azure.BlobAccess
+import gov.cdc.prime.router.fhirengine.translation.hl7.FhirToHl7Converter
+import gov.cdc.prime.router.fhirengine.translation.hl7.FhirTransformer
+import gov.cdc.prime.router.fhirengine.translation.hl7.schema.ConfigSchemaReader
+import gov.cdc.prime.router.fhirengine.translation.hl7.schema.converter.ConverterSchema
+import gov.cdc.prime.router.fhirengine.translation.hl7.schema.fhirTransform.FhirTransformSchema
+import gov.cdc.prime.router.fhirengine.translation.hl7.utils.ConstantSubstitutor
+import gov.cdc.prime.router.fhirengine.utils.FhirTranscoder
+import javax.ws.rs.InternalServerErrorException
+
+class TranslationSchemaManager {
+    enum class SchemaType(val directory: String) {
+        FHIR("fhir_transforms"),
+        HL7("hl7_mapping"),
+    }
+
+    /**
+     * Downloads the content of each blob in the [directory] with [blobContainerInfo] takes the input file, runs
+     * the transform on it, and compares it to the output validating that they are the same.
+     */
+    fun validateManagedSchemas(
+        schemaType: SchemaType,
+        blobContainerInfo: BlobAccess.BlobContainerMetadata = BlobAccess.defaultBlobMetadata,
+        blobEndpoint: String,
+    ): Boolean {
+        val sourceContainer = BlobAccess.getBlobContainer(blobContainerInfo)
+        val blobs =
+            BlobAccess.listBlobs(schemaType.directory, blobContainerInfo, false)
+        val inputs = blobs.filter { it.currentBlobItem.name.startsWith("input.") }
+
+        inputs.forEach { currentInput ->
+            val inputDirectoryPath = Regex("(.*/).*").find(currentInput.currentBlobItem.name)!!.groups[1]!!.value
+            val input = sourceContainer.getBlobClient(inputDirectoryPath + "input.fhir")
+            val inputBundle = FhirTranscoder.decode(input.toString())
+
+            val transformBlob = blobs.filter {
+                it.currentBlobItem.name.startsWith(inputDirectoryPath) &&
```

Comment:
I don' t think this logic will work. Looking at the design note, the structure shall look something like this:

```
/azure-blob-container
    /fhir_mapping
        valid-2023-03-29T23:53:00.488Z.txt
        previous-valid-2023-03-25T23:53:00.488Z.txt
    /fhir_transforms
        valid-2023-03-29T23:53:00.488Z.txt
        previous-valid-2023-03-25T23:53:00.488Z.txt
        /common
        /receivers
        /senders
    /hl7_mapping
        valid-2023-03-29T23:53:00.488Z.txt
        previous-valid-2023-03-25T23:53:00.488Z.txt
        /receivers
            /STLTS
                /CA
                    CA.yml
                    /resources
                        ca-aoe-note.yml
                        ...
                    input.fhir
                    output.hl7
            /flexion
                ...
```

IF we were to grab the input.fhir shown above, this logic will return multiple .yml files? Instead of always expecting the first result to be the one we want, may be better to match a regex that looks for exactly `inputDirectorypath/<someStringWithNoSlash>.yml`?